### DISCUSSION TITLE
Bilgisayarı Anlayalım

### DISCUSSION THREAD

**Discussion by nilbuyukkinaci** [sonsuzus/sonsuzus#393] [Soru Cevap]
Problem
Bay ABC yakın zamanda bilgisayarda bölme işlemini öğreniyordu. Bilgisayarın temel modelini göz önünde bulundurarak, bir X sayısını D'ye (yani X/D) bölmek ve sonucu elde etmek istediğimizi varsayalım (bunun integer bölümü olduğuna dikkat edin, yani 7/2'nin sonucu 3 olacaktır).

Bilgisayar şu durumlarda bölme overflowu hatası verecektir:

Ortaya çıkan sayının (bölüm) ikili gösterimindeki (başa sıfır eklemeden) bit sayısı, bölen (D)'nin (başa sıfır eklemeden) ikili gösterimindeki bit sayısından daha fazladır.

Şimdi Bay ABC'ye T test durumu için bir integer X verildi. Bölme overflowunun oluşmayacağı bu tür bölenlerin D(1 <= D <= X) sayısını bulmak istiyor.

Kısıtlamalar
1 ≤ T ≤ 10^6
1 ≤ X ≤ 10^12

Girdi Formatı
İlk satırda, test durumlarının sayısı olan T verilir.
Sonraki T satır boyunca, her satır bir X tamsayısını içerir.

Çıktı Formatı
Her test durumu için beklenen çıktıyı yeni satırda yazdırın.

Örnek Girdi
2
3
5

Örnek Çıktı
2
4

Soru Kaynağı: https://www.hackerearth.com/practice/basic-programming/operators/basics-of-operators/practice-problems/algorithm/let-us-understand-computer-78476e7a/

**Comment by EfeMertIrez** [comment]
```
import sys
def fun(i):
    s= int(i ** (0.5))
    l = len(bin(s)) - 3
    A= 2 ** (l+1)
    B= i //A
    A= (2 ** (l)) - 1
    m= max(A,B)
    print(i - m)
cases = sys.stdin.readlines()[1:]
for i in cases:
    fun(int(i))
```
